File: test/unit/core/helm/chart/ChartDependencyUpdateRequestTest.java
/*
 * Copyright (C) 2023 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.hedera.fullstack.helm.client.test.proxy.request.chart;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

import com.hedera.fullstack.helm.client.execution.HelmExecutionBuilder;
import com.hedera.fullstack.helm.client.proxy.request.chart.ChartDependencyUpdateRequest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
class ChartDependencyUpdateRequestTest {
    @Mock
    HelmExecutionBuilder helmExecutionBuilderMock;

    @Test
    @DisplayName("Verify ChartDependencyUpdateRequest apply")
    void testChartDependencyUpdateRequestApply() {
        final ChartDependencyUpdateRequest request = new ChartDependencyUpdateRequest("mocked");
        assertThat(request).isNotNull();
        assertThat(request.chartName()).isEqualTo("mocked");

        when(helmExecutionBuilderMock.subcommands("dependency", "update")).thenReturn(helmExecutionBuilderMock);
        when(helmExecutionBuilderMock.positional("mocked")).thenReturn(helmExecutionBuilderMock);
        request.apply(helmExecutionBuilderMock);
        verify(helmExecutionBuilderMock, times(1)).subcommands("dependency", "update");
        verify(helmExecutionBuilderMock, times(1)).positional("mocked");
    }
}

File: test/unit/core/helm/chart/chart-dependency-update-request.test.ts
// SPDX-License-Identifier: Apache-2.0

import {expect} from 'chai';
import sinon from 'sinon';
import {describe, it} from 'mocha';
import {type HelmExecutionBuilder} from '../../../../../src/integration/helm/execution/helm-execution-builder.js';
import {ChartDependencyUpdateRequest} from '../../../../../src/integration/helm/request/chart/chart-dependency-update-request.js';

describe('ChartDependencyUpdateRequest Tests', () => {
  it('Verify ChartDependencyUpdateRequest apply', () => {
    const helmExecutionBuilderMock = {
      subcommands: sinon.stub().returnsThis(),
      positional: sinon.stub().returnsThis(),
    } as unknown as HelmExecutionBuilder;

    const request = new ChartDependencyUpdateRequest('mocked');
    expect(request).to.not.be.null;
    expect(request.chartName).to.equal('mocked');

    request.apply(helmExecutionBuilderMock);

    expect(helmExecutionBuilderMock.subcommands).to.have.been.calledOnceWith('dependency', 'update');
    expect(helmExecutionBuilderMock.positional).to.have.been.calledOnceWith('mocked');
  });

  it('should throw error when chartName is blank', () => {
    expect(() => new ChartDependencyUpdateRequest('  ')).to.throw('chartName must not be blank');
  });
});
